src/headerssync.cpp

```diff
@@ -54,6 +78,7 @@ void HeadersSyncState::Finalize()
     Assume(m_state != State::FINAL);
     ClearShrink(m_header_commitments);
     m_presync_last_header_received.SetNull();
+    std::vector<CompressedHeader>{}.swap(m_headers_cache);
```

Comment:
this may bare a comment stating that when we detect any problem the cache is completely wiped.
And can this also be a `ClearShrink` instead?